src/test/java/com/example/postgresdemo/service/QuestionServiceTest.java

```diff
@@ -0,0 +1,124 @@
+package com.example.postgresdemo.service;
+
+import com.example.postgresdemo.exception.ResourceNotFoundException;
+import com.example.postgresdemo.model.Question;
+import com.example.postgresdemo.model.QuestionRequestDTO;
+import com.example.postgresdemo.model.QuestionResponseDTO;
+import com.example.postgresdemo.repository.QuestionRepository;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.springframework.data.domain.Page;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+import static org.mockito.ArgumentMatchers.any;
+
+@ExtendWith(MockitoExtension.class)
+class QuestionServiceTest {
+
+    @Mock
+    QuestionRepository questionRepository;
+
+    @InjectMocks
+    QuestionService questionService;
+
+    @Test
+    void testFindAll() {
+        Question question1 = new Question();
+        question1.setId(1L);
+        question1.setTitle("Title1" );
+        question1.setDescription("Description1");
+        Question question2 = new Question();
+        question2.setId(2L);
```

Comment:
Please assert that dtos have same ids to also cover id mapping code

Reply:
Done